src/main/java/org/spongepowered/api/item/recipe/Recipe.java

```diff
@@ -47,29 +44,18 @@
 public interface Recipe {
 
     /**
-     * Returns the list of item types that result when successful crafting of
-     * this Recipe is completed.
+     * Gets the result of this {@link Recipe}.
      *
-     * @return The resultant list of item types
+     * @return the results of this {@link Recipe}
      */
-    List<ItemType> getResultTypes();
+    Collection<ItemStackSnapshot> getResults();
 
     /**
-     * Checks if the given {@link GridInventory} fits the required constraints
-     * to craft this Recipe.
+     * Gets the {@link RecipeRegistry} that
+     * corresponds with this Recipe.
      *
-     * @param grid The ItemGrid to check for validity
-     * @return True if the given input matches this recipe's requirements
+     * @return the {@link RecipeRegistry} that corresponds with this Recipe
      */
-    boolean isValid(GridInventory grid);
-
-    /**
-     * Returns the results for running this Recipe over an {@link GridInventory}
-     *
-     * @param grid An ItemGrid as input
-     * @return A list of ItemStacks or {@link Optional#empty()} if the given
-     *          ItemGrid does not match this recipe's requirements.
-     */
-    Optional<List<ItemStack>> getResults(GridInventory grid);
+    RecipeRegistry<?> getRegistry();
```

Comment:
If this recipe is restricted to a single registry you should throw an IllegalArgumentException if you try to register it in a different one.

Is there a builder method for this one?